go/src/lib.rs

```diff
@@ -229,17 +230,35 @@ pub unsafe extern "C" fn free_connection_response(
 /// * `command_response_ptr` must be valid until `free_command_response` is called.
 /// * The contained `string_value` must be obtained from the `CommandResponse` returned in [`SuccessCallback`] from [`command`].
 /// * The contained `string_value` must be valid until `free_command_response` is called and it must outlive the `CommandResponse` that contains it.
+/// * The contained `array_elements_len` must be obtained from the `CommandResponse` returned in [`SuccessCallback`] from [`command`].
+/// * The contained `array_elements_len` must be valid until `free_command_response` is called and it must outlive the `CommandResponse` that contains it.
+/// * The contained `array_value` must be obtained from the `CommandResponse` returned in [`SuccessCallback`] from [`command`].
+/// * The contained `array_value` must be valid until `free_command_response` is called and it must outlive the `CommandResponse` that contains it.
 #[no_mangle]
 pub unsafe extern "C" fn free_command_response(command_response_ptr: *mut CommandResponse) {
     assert!(!command_response_ptr.is_null());
     let command_response = unsafe { Box::from_raw(command_response_ptr) };
     let string_value = command_response.string_value;
     let string_value_len = command_response.string_value_len;
+    let array_value = command_response.array_value;
+    let array_value_len = command_response.array_value_len;
+    let array_elements_len = command_response.array_elements_len;
     drop(command_response);
     if !string_value.is_null() {
         let len = string_value_len as usize;
         unsafe { Vec::from_raw_parts(string_value, len, len) };
     }
+    if !array_value.is_null() {
+        let len = array_value_len as usize;
+        let vec = unsafe { Vec::from_raw_parts(array_value, len, len) };
+        let elem_len = unsafe { Vec::from_raw_parts(array_elements_len, len, len) };
+        for i in 0..len {
```

Comment:
Just to confirm that there is no memory leaks

Reply:
We are freeing up all the memory here which was allocated while converting the RedisValue to Vector. So, there should be no memory leaks.